cpp/src/cylon/net/ucc/ucc_operations.hpp

```diff
@@ -70,6 +70,11 @@ class UccTableGatherImpl : public net::TableGatherImpl {
   Status WaitAll(int32_t num_buffers) override;
 
  private:
+  // the following three are to mimic gather using allgather
+  std::vector<std::vector<int>>* displacements_;
+  std::vector<std::vector<int>>* all_recv_counts_;
```

Comment:
I have a feeling that you might not even need these. I've explained it in a previous comment